cmd/main.go

```diff
@@ -26,29 +31,46 @@ func main() {
 	// Query the OIDC provider
 	config.SetOidcProvider()
 
+	authenticator := authentication.NewAuthenticator(config)
 	// Get clientset for Authorizers
-	var clientset kubernetes.Interface
-	if config.EnableRBAC {
-		icc, err := rest.InClusterConfig()
-		if err != nil {
-			log.Fatalf("error getting in cluster configuration for RBAC client: %v", err)
-		}
-		clientset, err = kubernetes.NewForConfig(icc)
+	var clientset k8s.Interface
+	if config.EnableRBAC || config.EnableInClusterStorage {
+		var err error
+		clientset, err = kubernetes.GetClientSet()
 		if err != nil {
 			log.Fatalf("error getting kubernetes client: %v", err)
 		}
-	} else {
-		clientset = nil
 	}
 
-	// Prepare cookie session store (first key is for auth, the second one for encryption)
-	cookieStore := sessions.NewCookieStore(config.Secret, []byte(config.SessionKey))
-	cookieStore.Options.MaxAge = int(config.Lifetime / time.Second)
-	cookieStore.Options.HttpOnly = true
-	cookieStore.Options.Secure = !config.InsecureCookie
+	var userInfoStore v1alpha1.UserInfoInterface
+	if !config.EnableInClusterStorage {
+		// Prepare cookie session store (first key is for auth, the second one for encryption)
+		cookieStore := sessions.NewCookieStore(config.Secret, []byte(config.SessionKey))
+		cookieStore.Options.MaxAge = int(config.Lifetime / time.Second)
+		cookieStore.Options.HttpOnly = true
+		cookieStore.Options.Secure = !config.InsecureCookie
+
+		userInfoStore = &storage.GorillaUserInfoStore{
+			SessionStore: cookieStore,
+			SessionName:  config.ClaimsSessionName,
+		}
+	} else {
+		clusterStorage := cluster.NewClusterStore(
+			clientset,
+			config.ClusterStoreNamespace,
+			string(config.Secret),
+			config.Lifetime,
+			time.Duration(config.ClusterStoreCacheTTL)*time.Second,
+			authenticator)
+
+		gc := cluster.NewGC(clusterStorage, time.Minute, false, true)
 
```

Comment:
```suggestion
		userInfoStore := cluster.NewClusterStore(
			clientset,
			config.ClusterStoreNamespace,
			string(config.Secret),
			config.Lifetime,
			time.Duration(config.ClusterStoreCacheTTL)*time.Second,
			authenticator)

		gc := cluster.NewGC(userInfoStore, time.Minute, false, true)

```

Reply:
great catch! I found this while running e2e tests in konvoy, and somehow missed this comment :)